simulation/test_socks5.cpp

```diff
@@ -220,7 +220,7 @@ TORRENT_TEST(socks5_tcp_listen_alert)
 		[&](lt::session&, lt::alert const* alert) {
 			if (auto* a = lt::alert_cast<lt::listen_succeeded_alert>(alert))
 			{
-				if (a->sock_type == listen_succeeded_alert::socks5)
+				if (a->socket_type == listen_succeeded_alert::socks5)
```

Comment:
If you're going to change this name (i.e. the API), what do you think about changing its type as well, to be a proper enum class? (that way it doesn't have to be a member of listen_succeeded_alert either)